### DISCUSSION TITLE
Recover from RBF on Channel Open, Confirmed On-Chain, Stuck Pending Old TXN

### DISCUSSION THREAD

**Discussion by atomantic** [lightningnetwork/lnd#8208] [Troubleshooting] [answered]
Edit: This problem is from me doing something stupid, so hopefully this thread will aid anyone else who does the same. Somehow I landed on a thread somewhere that suggested simply broadcasting an RBF transaction would be good enough--and I missed the bit in the LND docs that say you can't do that for channel opens.

The situation:

- I had a pending channel open with a fee that was too low for the demand period (it wasn't really very low)
- I decided I should RBF to increase the fee and get it through so I manually broadcasted this onchain without using the lncli bumpfee tools
- Now I have a successful transaction, confirmed dozens on times onchain, but a pending open that is still looking at the txn id of the pre-RBF txn.

**Comment by atomantic** [comment]
ok, after reading more of the LND guide, I see that RBF is not a valid way to increase fees on a channel open.
Chantools guide says I should NOT abandonchannel since this txn is confirmed onchain.
What's the best way to close this out and recover?

**Reply by guggero** [reply]
Correct, using RBF is the worst thing you can do with a channel funding transaction! **Only** CPFP is allowed (that's what `lncli wallet bumpfee` does).
The reason is this: When you open a channel, you send funds to a 2-of-2 multisig address to which your node only has one key. Since that's dangerous, as you need cooperation of your peer to get the funds out, your wallet will only sign that transaction once it has a counter-signed commitment transaction that spends those funds again and reflects the current initial channel balance.
But that commitment transaction references the TXID of a specific funding transaction. If you publish another funding transaction with a different TXID, you basically invalidate the commitment transaction.

So fixing this isn't just a case of updating an entry in the database. You need a new signature from your peer. You can use the `chantools rescuefunding` (and your peer `chantools signrescuefunding`).

**Reply by atomantic** [reply]
Thanks @guggero -- I'm getting an error when running `chantools rescuefunding`:

```
error fetching UTXO info for outpoint MY_CHANNEL_POINT: transaction not found
```

Looks like the default API of blockstream doesn't persist the info for the RBF source. I tried setting --apiurl "https://mempool.space/api" -- I see the txn on mempool. But I'm not finding it there either...

Do you know another API source that might work?

**Reply by atomantic** [reply]
Ah, nevermind. I missed supplying the --confirmedchannelpoint

Great tool! I'll post a full recovery summary for posterity once complete :)

**Comment by atomantic** [comment, marked as answer]
Alright. For anyone else who somehow manages to do what I did, here's how to resolve it when running on an Umbrel node:

1. Edit `~/umbrel/app-data/data/lnd/umbrel-lnd.conf` to add `reset-wallet-transactions=true`
2. Restart umbrel (this will rescan the wallet to fix UTXO sync issue and may take a while after restart) — you can watch along via ssh into umbrel and sudo docker logs —follow ${the docker id of the lnd container}
3. [Install chantools](https://github.com/lightninglabs/chantools) -- note: I installed this on my host machine rather than installing it on the umbrel
4. Backup lightning db from umbrel to local host so we don’t have to stop the lnd service to use the db: `scp umbrel@umbrel.local:~/umbrel/app-data/lightning/data/lnd/data/graph/mainnet/channel.db channel.db`
5. get a new bitcoin onchain deposit address from your node (DEPOSIT_ADDR)
6. Use chantools [rescuefunding](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_rescuefunding.md): 
```
chantools rescuefunding --channeldb ./channel.db --dbchannelpoint OLD_TXN_ID:0 --confirmedchannelpoint NEW_RBF_TXN_ID:0 --sweepaddr DEPOSIT_ADDR
```
7. Contact the peer and have them use [chantools signrescuefunding](https://github.com/lightninglabs/chantools/blob/master/doc/chantools_signrescuefunding.md), providing them with the output from  above: 
```
chantools signrescuefunding --psbt YOUR_PSBT
```
11. Broadcast signed transaction
12. Kill the pending channel: 
```
sudo ~/umbrel/scripts/app compose lightning exec lnd lncli abandonchannel --chan_point OLD_TXN_ID:0 --i_know_what_i_am_doing
```
13. open a new channel to the peer


Final question for you @guggero: Does the peer also have to run abandonchannel to clear their db of the bad open from my node or will my abandonment convey across the network? I ask because they have their `maxpendingchannels` set to 1 and it appears my attempts to make a new connection are blocked because of that.

**Reply by guggero** [reply]
Yes, the peer can run the `abandonchannel` command. Or they can wait for 2 weeks, the responder side to a channel open is supposed to discard a channel if the funding transaction doesn't confirm within 2 weeks.

**Reply by bushev** [reply]
Hey guys, I have an identical situation where I accidentally created a funding output for a channel that never resulted in a proper channel, and no commitment transactions exist. However, in my case, the other party (the node I was opening the channel with) is not cooperative. They cannot execute any commands on their side and outright refuse to do anything.

Am I correct in understanding that my funds are permanently stuck and can never be recovered without the other party’s signature?

**Reply by guggero** [reply]
Yeah, unfortunately that's the case. If there is no valid commitment transaction available, then you basically have funds in a 2-of-2 multisig output to which you only have one key.
Without cooperation there's nothing you can do. Why does no commitment transaction exist in your case?

**Reply by bushev** [reply]
The other party, the remote node, is a liquidity provider with many other clients. They explained that they cannot install additional software or use third-party tools to sign the transaction. Specifically, they refuse to run the `chantools signrescuefunding` command. They’ve stated that they don’t want to take risks with their security or operations for this.

Unfortunately, I have nothing to offer them as an incentive, given that the amount of my transaction is around $500. It’s unlikely they would be interested in splitting such a small sum, and I understand their reasoning. However, it feels frustrating because this situation leaves my funds effectively stuck.

If there were a simpler way to resolve this — one that doesn’t involve additional tools and allows them to straightforwardly sign a Partially Signed Transaction — it might be more acceptable to them. But as far as I understand, no such option exists, leaving me at an impasse...

**Reply by yyforyongyu** [reply]
i think `lnd` would never broadcast the funding tx unless there's an initial commitment tx created and signed. So I'm curious did you use `lnd` to create the channel?

**Reply by bushev** [reply]
Yes I do, my stack is:

- LND 0.18.3-beta
- ThunderHub 0.13.31

Unfortunately, the LND logs didn’t survive in Docker after the last restart. So, unfortunately, I can’t go back to the logs to check exactly what happened anymore.

**Reply by guggero** [reply]
But how did you end up with a channel that you don't have a commitment transaction for? Are you sure you don't have one? Did you delete the database? What do you see in `lncli pendingchannels`?

**Reply by bushev** [reply]
Yes, I have a complete backup of my channels (`channel.db` file). At the time of the issue, there were only two channels. I also have the output of the command `lncli exportchanbackup --all`.

![image](https://github.com/user-attachments/assets/8ad47b27-e9df-4cdf-a3b5-9fa513b28115)

After encountering the issue, I deleted the database file and restored it to try and force the channels to close automatically. I remember reading this advice somewhere, possibly from ChatGPT. Following this process, one channel — the valid one — successfully closed without any problems. However, the other channel, which couldn’t open properly, entered a Pending Close status and has remained in this state ever since.

I recently listed the pending channels, but unfortunately, I don’t see any associated transactions for the problematic channel. I also attempted to open the channel.db file using the bbolt Utility, but I haven’t made significant progress in extracting or analyzing the data yet.

```
lncli pendingchannels
{
    "total_limbo_balance":  "0",
    "pending_open_channels":  [],
    "pending_closing_channels":  [],
    "pending_force_closing_channels":  [],
    "waiting_close_channels":  [
        {
            "channel":  {
                "remote_node_pub":  "0<redacted>3",
                "channel_point":  "5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1",
                "capacity":  "556041",
                "local_balance":  "0",
                "remote_balance":  "0",
                "local_chan_reserve_sat":  "0",
                "remote_chan_reserve_sat":  "0",
                "initiator":  "INITIATOR_LOCAL",
                "commitment_type":  "ANCHORS",
                "num_forwarding_packages":  "0",
                "chan_status_flags":  "ChanStatusRestored",
                "private":  true,
                "memo":  ""
            },
            "limbo_balance":  "0",
            "commitments":  {
                "local_txid":  "",
                "remote_txid":  "",
                "remote_pending_txid":  "",
                "local_commit_fee_sat":  "0",
                "remote_commit_fee_sat":  "0",
                "remote_pending_commit_fee_sat":  "0"
            },
            "closing_txid":  "",
            "closing_tx_hex":  ""
        }
    ]
}
```

**Reply by guggero** [reply]
Did you ever try to RBF that problematic channel manually? If not, then it just looks like the channel never opened (as the transaction doesn't exist on chain), so the funds were never locked, meaning you don't have this situation after all (just a normal channel that never made it to the chain and you should have the funds back in your wallet).

**Reply by bushev** [reply]
Here’s what happened. I started opening a channel using ThunderHub, completed all the required steps, and left for a while. When I came back a few hours later, I noticed that the channel hadn’t been opened to my initial tx (not longer exists - replaced): 5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077

I checked the transaction in the mempool and saw that it was still unconfirmed because the fee was very, very low. I thought, “Okay, I can use the Replace-By-Fee feature in Sparrow Wallet to replace this transaction with one that has a higher fee.”

To manage my UTXOs more easily, I connected Sparrow Wallet to my Lightning Node. Then I performed the Replace-By-Fee through Sparrow Wallet and got a new transaction, which confirmed quickly (https://mempool.space/tx/55f1d8e4ba03de62eaf3bbdc7a3f1ed376b9f1a30bd6391f2cf45f1388c13461). After that, nothing else happened. The channel was never opened, and no further changes occurred.

Since then, I’ve tried deleting the database, restoring it, and doing various movements to resolve the issue, but none of that has worked or led to any results.

![image](https://github.com/user-attachments/assets/8b64d441-a496-49cc-b2e1-d0d011ae39cf)

```
2025-01-26 12:26:24.560 [INF] PEER: Peer(02b4552a7a85274e4da01a7c71ca57407181752e8568b31d51f13c111a2941dce3): Loading ChannelPoint(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1), isPending=false
2025-01-26 12:26:24.560 [WRN] PEER: Peer(02b4552a7a85274e4da01a7c71ca57407181752e8568b31d51f13c111a2941dce3): Unable to find our forwarding policy for channel 5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1, using default values
2025-01-26 12:26:24.560 [INF] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): stopping
2025-01-26 12:26:24.560 [WRN] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): error when syncing channel states: link shutting down
2025-01-26 12:26:24.560 [INF] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): exited
2025-01-26 12:26:24.560 [INF] HSWC: Removing channel link with ChannelID(7760a19b5217e0d6309c1e6cd975f274a83f6ba6d43658fea9335945c1d78758)
2025-01-26 12:26:24.560 [INF] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): starting
2025-01-26 12:26:24.560 [INF] HSWC: Adding pending link chan_id=7760a19b5217e0d6309c1e6cd975f274a83f6ba6d43658fea9335945c1d78758, short_chan_id=0:0:0
2025-01-26 12:26:24.561 [INF] PEER: Peer(02b4552a7a85274e4da01a7c71ca57407181752e8568b31d51f13c111a2941dce3): Negotiated chan series queries
2025-01-26 12:26:24.561 [INF] DISC: Creating new GossipSyncer for peer=02b4552a7a85274e4da01a7c71ca57407181752e8568b31d51f13c111a2941dce3
2025-01-26 12:26:24.561 [WRN] LNWL: ChannelPoint(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): Set available amount to 0 because we could not pay for the CommitmentFee of the new ChannelState: ourBalance is negative after applying the fee: ourBalance=0, current commitFee(w/o additional htlc)=0 mSAT, feeBuffer=0 mSAT (type=feebuffer) local_chan_initiator
2025-01-26 12:26:24.561 [WRN] LNWL: ChannelPoint(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): Set available amount to 0 because we could not pay for the CommitmentFee of the new ChannelState: ourBalance is negative after applying the fee: ourBalance=0, current commitFee(w/o additional htlc)=0 mSAT, feeBuffer=0 mSAT (type=feebuffer) local_chan_initiator
2025-01-26 12:26:24.561 [INF] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): HTLC manager started, bandwidth=0 mSAT
2025-01-26 12:26:24.561 [INF] HSWC: ChannelLink(5987d7c1455933a9fe5836d4a66b3fa874f275d96c1e9c30d6e017529ba16077:1): Attempting to re-synchronize channel: SCID=0:0:0, status=ChanStatusRestored, initiator=true, pending=false, local commitment has height=0, local_htlc_index=0, local_log_index=0, remote_htlc_index=0, remote_log_index=0, remote commitment has height=0, local_htlc_index=0, local_log_index=0, remote_htlc_index=0, remote_log_index=0

```

**Reply by guggero** [reply]
Ah, okay. Yeah, then unfortunately the `chantools zombierecovery` is the only way to get those funds back (requiring the cooperation of the peer)...

**Reply by bushev** [reply]
Got it, thank you so much for taking the time to help and explain everything. I really appreciate it. At this point, I think it’s time for me to let this one go and move on. Sometimes you just have to accept the situation and learn from it. Thanks again for your efforts, and I’ll focus on moving forward from here.
